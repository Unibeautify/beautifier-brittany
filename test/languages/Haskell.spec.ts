import * as fs from "fs";
import * as path from "path";

import { newUnibeautify, Beautifier } from "unibeautify";
import beautifier from "../../src";
import { raw } from "../utils";

jest.setTimeout(10000);
describe("should successfully beautify Haskell files", () => {
  // tslint:disable:mocha-no-side-effect-code
  testFile("test.hs");
});

function testFile(fixtureFileName: string) {
  test(`should successfully beautify file ${fixtureFileName}`, () => {
    const text: string = fs
      .readFileSync(path.resolve(__dirname, `../fixtures/${fixtureFileName}`))
      .toString();
    const unibeautify = newUnibeautify();
    unibeautify.loadBeautifier(beautifier);
    return unibeautify
      .beautify({
        languageName: "Haskell",
        options: {},
        text,
      })
      .then(results => {
        expect(raw(results)).toMatchSnapshot();
      });
  });
}
