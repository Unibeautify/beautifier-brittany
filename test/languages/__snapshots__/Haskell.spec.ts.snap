// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should successfully beautify Haskell files should successfully beautify file test.hs 1`] = `
Object {
  Symbol(raw): "-- https://wiki.haskell.org/The_Fibonacci_sequence#Monadic
import Control.Monad.State
fib n = flip evalState (0, 1) $ do
  forM [0 .. (n - 1)] $ \\\\_ -> do
    (a, b) <- get
    put (b, a + b)
  (a, b) <- get
  return a
",
}
`;
